bzt/resources/selenium_extras.py

```diff
@@ -52,6 +95,111 @@ def _send_exit_flow_marker(*args, **kwargs):   # for apiritif. remove when compa
     _send_marker(stage, params)
 
 
+def dialogs_replace():
+    """
+    Replaces the standard JavaScript methods, i.e. 'window.confirm', 'window.alert' and 'window.prompt' with
+    own implementation that stores the messages from the dialogs and also is capable of returning user defined
+    values
+    """
+
```

Comment:
@havja @greyfenrir why this 'if' missed in the refactoring? Now i'm getting this execute script command after each `driver.url` command. Is it expected behaviour?
``` 
       if not self.is_active:
            return  # don't replace dialogs in case DialogsManager is not activated
```

Reply:
Hi, yes it is an expected behaviour. We need to call this script after each change of url or any navigation in order to make sure that the JS window methods are replaced as we need otherwise the dialogs replacement logic will not work. This logic is mostly taken out of the Selenium implementations in Java and they do the same thing there...